apps/retrospective/services/sprint_sync.go

```diff
@@ -494,13 +495,30 @@ func (service SprintService) changeTaskEstimates(tx *gorm.DB, task retroModels.T
 }
 
 func (service SprintService) addOrUpdateTaskTrackerTask(
-	sprintID uint,
+	sprint retroModels.Sprint,
 	ticket taskTrackerSerializers.Task,
 	retroID uint,
 	alternateTaskKey string) (err error) {
-	tx := service.DB.Begin()
 
+	db := service.DB
+
+	DoneStatusList, err := tasktracker.GetDoneStatusMapping(sprint.Retrospective.TaskProviderConfig)
```

Comment:
Handle the case when the task provider config doesn't have the key for the done status mapping (existing retrospectives).